book/src/operations.md

```diff
@@ -1 +1,29 @@
 # Operations
+The mechanism by which statements are derived is furnished by *operations*. Roughly speaking, with few exceptions, an operation deduces a statement from one or more existing statements according to some relation that must be satisfied between these statements. For example, if `Equal(ak1, ak2)` holds true, then the operation `SymmetricEq` applied to this statement yields `Equal(ak2, ak1)`.
+
+More precisely, an operation is a code (or, in the frontend, string identifier) followed by 0 or more arguments. These arguments may consist of up to three statements, up to one key-value pair and up to one Merkle proof.
+
+The following table summarises the natively-supported operations:
+
+| Code | Identifier         | Args                | Condition                                                                                                             | Output                                                         |
+|------|--------------------|---------------------|-----------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------|
+| 0    | `None`             |                     |                                                                                                                       | `None`                                                         |
+| 1    | `NewEntry`         | `(key, value)`      |                                                                                                                       | `ValueOf(ak, value)`, where `ak` has key `key` and origin ID 1 |
```

Comment:
This one is interesting.  The operation includes the `(key, value)` in its arguments so that it can generate a statement with that `key` and `value`.
If we think about the circuit, what is public is the statement and not the operation.  So we need the `key, value` be part of the `ValueOf` statement.  Then, in the circuit, do we need the `key, value` in the operation?  I don't think so.  The `key, value` in the `ValueOf` is just unconstrained witness data, it doesn't come from anywhere.

So it seems to me that in the language level a `NewEntry` has arguments `(key, value)` but in the circuit `NewEntry` doesn't have any argument.